src/java.desktop/share/classes/javax/swing/border/LineBorder.java

```diff
@@ -144,73 +143,34 @@ public LineBorder(Color color, int thickness, boolean roundedCorners)  {
+    }
+
+    private void paintUnscaledBorder(Component c, Graphics g,
+                                     int w, int h, double scaleFactor) {
         if ((this.thickness > 0) && (g instanceof Graphics2D)) {
             Graphics2D g2d = (Graphics2D) g;
-
-            AffineTransform at = g2d.getTransform();
-
-            // if m01 or m10 is non-zero, then there is a rotation or shear
-            // or if no Scaling enabled,
-            // skip resetting the transform
-            boolean resetTransform = ((at.getShearX() == 0) && (at.getShearY() == 0)) &&
-                    ((at.getScaleX() > 1) || (at.getScaleY() > 1));
-
-            int xtranslation;
-            int ytranslation;
-            int w;
-            int h;
-            int offs;
-
-            if (resetTransform) {
-                /* Deactivate the HiDPI scaling transform,
-                 * so we can do paint operations in the device
-                 * pixel coordinate system instead of the logical coordinate system.
-                 */
-                g2d.setTransform(new AffineTransform());
-                double xx = at.getScaleX() * x + at.getTranslateX();
-                double yy = at.getScaleY() * y + at.getTranslateY();
-                xtranslation = clipRound(xx);
-                ytranslation = clipRound(yy);
-                w = clipRound(at.getScaleX() * width + xx) - xtranslation;
-                h = clipRound(at.getScaleY() * height + yy) - ytranslation;
-                offs = this.thickness * (int) at.getScaleX();
-            } else {
-                w = width;
-                h = height;
-                xtranslation = x;
-                ytranslation = y;
-                offs = this.thickness;
-            }
-
-            g2d.translate(xtranslation, ytranslation);
-
+            int offs = this.thickness * (int) scaleFactor;
             Color oldColor = g2d.getColor();
             g2d.setColor(this.lineColor);
 
             Shape outer;
             Shape inner;
-
             int size = offs + offs;
             if (this.roundedCorners) {
                 float arc = .2f * offs;
                 outer = new RoundRectangle2D.Float(0, 0, w, h, offs, offs);
                 inner = new RoundRectangle2D.Float(offs, offs, w - size, h - size, arc, arc);
-            }
-            else {
+            } else {
                 outer = new Rectangle2D.Float(0, 0, w, h);
                 inner = new Rectangle2D.Float(offs, offs, w - size, h - size);
             }
+
             Path2D path = new Path2D.Float(Path2D.WIND_EVEN_ODD);
             path.append(outer, false);
             path.append(inner, false);
             g2d.fill(path);
-            g2d.setColor(oldColor);
-
-            g2d.translate(-xtranslation, -ytranslation);
-
-            if (resetTransform) {
-                g2d.setTransform(at);
-            }
+            g.setColor(oldColor);
```

Comment:
```suggestion

            g2d.setColor(oldColor);
```
Use `g2d` as it was originally and add a blank line to separate fill — painting the border — from the cleanup, restoring the color.